editor/server_api.cpp

```diff
@@ -0,0 +1,125 @@
+#include "editor/server_api.hpp"
+
+#include "coding/url_encode.hpp"
+
+#include "geometry/mercator.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "std/sstream.hpp"
+
+namespace osm
+{
+
+ServerApi06::ServerApi06(OsmOAuth const & auth)
+  : m_auth(auth)
+{
+}
+
+bool ServerApi06::CreateChangeSet(TKeyValueTags const & kvTags, uint64_t & outChangeSetId) const
+{
+  if (!m_auth.IsAuthorized())
+  {
+    LOG(LWARNING, ("Not authorized"));
+    return false;
+  }
+
+  ostringstream stream;
+  stream << "<osm>\n"
+  "<changeset>\n";
+  for (auto const & tag : kvTags)
+    stream << "  <tag k=\"" << tag.first << "\" v=\"" << tag.second << "\"/>\n";
+  stream << "</changeset>\n"
+  "</osm>\n";
+
+  OsmOAuth::Response const response = m_auth.Request("/changeset/create", "PUT", move(stream.str()));
+  if (response.first == OsmOAuth::ResponseCode::OK)
+  {
+    if (strings::to_uint64(response.second, outChangeSetId))
+      return true;
+    LOG(LWARNING, ("Can't parse changeset ID from server response."));
+  }
+  else
+    LOG(LWARNING, ("CreateChangeSet request has failed:", response.first));
+
+  return false;
+}
+
+bool ServerApi06::CreateNode(string const & nodeXml, uint64_t & outCreatedNodeId) const
+{
+  OsmOAuth::Response const response = m_auth.Request("/node/create", "PUT", move(nodeXml));
```

Comment:
move is useless here, as nodeXml is const reference. One can not simple steal internals from const references.